src/compile.rs

```diff
@@ -387,11 +388,11 @@ impl Compiler {
         } else {
             let entry = self.insts.len();
             let hole = self.push_hole(InstHole::Save { slot: first_slot });
-            let patch = self.c(expr)?.unwrap_or(self.next_inst());
+            let patch = self.c(expr)?.unwrap_or_else(|| self.next_inst());
```

Comment:
Since the same kind of pattern (`.unwrap_or_else(|| self.next_inst()))`) is repeated  at least 4 times, it might be better to introduce a dedicated function? I'll try to look into it

Reply:
Possibly.

To be honest, this whole compiler is going to be thrown away "soon" as part of #656. The current state of that rewrite (not quite to parity yet) is here: https://github.com/BurntSushi/regex-automata/blob/c4fee38d7499721a2eccaa7074f21bf2c955cbce/src/nfa/thompson/compiler.rs

The rewrite is faster for the cases I've tested, generates more compact bytecode and is overall much clearer. (It does away with the whole `InstHole` thing.)

So I think it's probably not wise to spend too much time refactoring or clarifying the code here. :-) With that said, a helper function sounds fine if you want to pursue that.

Reply:
I have kept the `unwrap_or_else` for those cases, I hope that's ok!